Chap_Introduction.tex

```diff
@@ -232,4 +234,81 @@ \subsection{PMIx Roles}
 These \emph{client-role,server-role, and tool-role \ac{PMIx} standard compliant} implementations have the advantage of being able to support a broad set of \ac{PMIx} consumers in the different roles.
 \adviceimplend
 
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\subsection{Application Binary Interface (ABI)}
+\label{chap:intro:abi}
+
+An \acf{API} defines how data types, data structures, and functions are represented in source code.
+An \acf{ABI} defines how data types, data structures, and functions are represented in machine code for a given system making it platform-specific.
+An important aspect of an \ac{ABI} is the size, layout, and alignment of data structures.
+A stable \ac{ABI} may allow a program compiled with one implementation of the PMIx Standard to run with a different implementation of the PMIx Standard as long as both implementations adhere to the same \ac{ABI}.
+
+The PMIx Standard strives to maintain a stable \ac{ABI} to support applications and tools that rely on more than one implementation of the PMIx Standard.
+To facilitate such interoperability the PMIx \ac{ASC} maintains a set of standardized headers that are versioned with the PMIx Standard that applications and tools can reference~\footnote{PMIx Headers for ABI Compatibility \url{https://github.com/pmix/pmix-abi}}.
+
+In recognition that there are circumstances where the \ac{ABI} needs to be modified this section defines some guidance for making such modifications.
+Additions to the PMIx interface can occur without breaking \ac{ABI} compatibility.
+Deprecating and removing portions of the PMIx interface does break \ac{ABI} compatibility.
+Modifications to the existing PMIx interface do break \ac{ABI} compatibility.
+Any changes to the \ac{ABI} will occur in a major release and will be noted in the release notes.
+
+The PMIx \ac{ABI} is comprised of the following:
+\begin{compactitemize}
+\item Function signatures
+\item Constants and their values
+\item Structures and their membership (including field position)
+\item Attributes that are required to be implemented and their string representation
+\end{compactitemize}
+
+If a data structure must be changed in a way that breaks the \ac{ABI} the authors should consider creating a new data structure if possible.
+Additionally, the authors must take care to update or duplicate existing PMIx interfaces that are impacted by the change.
+
+Attributes that are required by the PMIx standard are a required part of the \ac{ABI}.
```

Comment:
Ralph comment: Attributes are non-linkable string objects (i.e., they are defines, they don't have an address as a variable), hence they are not an ABI element.

Reply:
```Although the values of attributes and constants are not part of the \ac{ABI}, they are defined as standardized members of the \ac{API}, hence their value can be assumed based on the \ac{ABI} version.```

Reply:
Please take time before Wednesday afternoon to suggest any revisions to this text.

This text will replace the full paragraph here (lines 266-271).

Reply:
I captured this change in the https://github.com/pmix/pmix-standard/pull/375/files#r870587051
Let's follow up on that comment.